File: CodebaseTechnologyScanner.API/Models/PaginatedResponse.cs
namespace CodebaseTechnologyScanner.API.Models;

public class PaginatedResponse<T>
{
    public List<T> Results { get; set; } = new();
    public PaginationMetadata Pagination { get; set; } = new();
}

public class PaginationMetadata
{
    public int CurrentPage { get; set; }
    public int PageSize { get; set; }
    public int TotalResults { get; set; }
    public int TotalPages { get; set; }
    public bool HasNextPage { get; set; }
    public bool HasPreviousPage { get; set; }
}

File: codebase-scanner-ui/src/models/PaginatedResponse.ts
import type { ScanResult } from './ScanResult';

export interface PaginatedResponse {
  results: ScanResult[];
  pagination: PaginationMetadata;
}

export interface PaginationMetadata {
  currentPage: number;
  pageSize: number;
  totalResults: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
